tests/system/src/test/java/test/com/sun/marlin/DashedRectTest.java

```diff
@@ -0,0 +1,217 @@
+    // the first test, and it lives through the execution of all tests.
+    public static class MyApp extends Application {
+
+        Stage stage = null;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            DashedRectTest.myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            this.stage = primaryStage;
+
+            // Corrupt Marlin Dasher.dash cached array:
+            final Path path1 = new Path();
+            path1.getElements().addAll(
+                    new MoveTo(20, 20),
+                    new HLineTo(70),
+                    new VLineTo(70),
+                    new HLineTo(20),
+                    new ClosePath()
+            );
+            path1.setStroke(Color.RED);
+            path1.setFill(null);
+            path1.setStrokeWidth(2);
+            path1.setStrokeDashOffset(DASH_PH);
+
+            final ObservableList<Double> pDashes = path1.getStrokeDashArray();
+            pDashes.clear();
+            for (int i = 0; i < 100; i++) {
+                pDashes.add(19.333);
+            }
+
+            // Create 2nd path shape
+            final Path path2 = new Path();
+            path2.getElements().addAll(
+                    new MoveTo(5, 5),
+                    new HLineTo(MAX),
+                    new VLineTo(MAX),
+                    new HLineTo(5),
+                    new ClosePath()
+            );
+            path2.setFill(null);
+            path2.setStroke(Color.BLUE);
+            path2.setStrokeWidth(2);
+            path2.setStrokeDashOffset(DASH_PH);
+            path2.getStrokeDashArray().setAll(DASH_LEN);
+
+            Scene scene = new Scene(new Group(path1, path2));
+            stage.setScene(scene);
+            stage.setTitle("DashedRectTest");
+            stage.show();
+
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        // Start the Application
+        new Thread(() -> Application.launch(MyApp.class, (String[]) null)).start();
+
+        try {
+            if (!launchLatch.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for Application to launch");
+            }
+
+        } catch (InterruptedException ex) {
+            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
+            err.initCause(ex);
+            throw err;
+        }
+
+        assertEquals(0, launchLatch.getCount());
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Platform.exit();
+    }
+
+    @Test(timeout = 10000)
+    public void TestDashedPath() throws InterruptedException, IOException {
+
+        final int size = MAX * 2;
+
+        final WritableImage img = new WritableImage(size, size);
+
+        done = false;
+        Platform.runLater(() -> {
```

Comment:
Util.runAndWait is preferred, since it takes care of both the waiting and the propagation of exceptions.